diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee08072..3aacddf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,7 +37,6 @@ if(shared)
 endif()
 
 add_subdirectory(libdwarf)
-add_subdirectory(dwarfdump)
 if ( BUILD_DWARFGEN )
 	if ( DWARF_WITH_LIBELF )
 	    add_subdirectory(dwarfgen)
@@ -50,4 +49,3 @@ if ( BUILD_DWARFEXAMPLE )
 endif()
 
 message(STATUS "Install prefix ... ${CMAKE_INSTALL_PREFIX}" )
-add_custom_target(dd DEPENDS ${DWARF_TARGETS} dwarfdump)
diff --git a/libdwarf/CMakeLists.txt b/libdwarf/CMakeLists.txt
index 9457b6d..a93ce05 100644
--- a/libdwarf/CMakeLists.txt
+++ b/libdwarf/CMakeLists.txt
@@ -81,23 +81,18 @@ foreach(i RANGE ${targetCount})
 	msvc_posix(${target})
 
 	target_link_libraries(${target} PUBLIC ${LIBELF_LIBRARIES})
-	
 	set_target_properties(${target} PROPERTIES OUTPUT_NAME dwarf)
 
-	set(SUFFIX $<$<EQUAL:${CMAKE_SIZEOF_VOID_P},8>:64>)
-	set(LIBDIR lib${SUFFIX})
-	set(BINDIR bin${SUFFIX})
-
 	install(TARGETS ${target}
-		RUNTIME DESTINATION ${BINDIR}		
-		LIBRARY DESTINATION ${LIBDIR}
-		ARCHIVE DESTINATION ${LIBDIR})
+		RUNTIME DESTINATION bin
+		LIBRARY DESTINATION lib
+		ARCHIVE DESTINATION lib)
 endforeach()
 
-if(UNIX AND shared)
-	target_link_libraries(dwarf-shared PUBLIC z)
-endif()
-		
+target_link_libraries(dwarf-shared PUBLIC z)
+
+install(FILES ${CMAKE_BINARY_DIR}/source_subfolder/libdwarf/libdwarf.h DESTINATION include)
+install(FILES dwarf.h DESTINATION include)
 
 if (DO_TESTING)
     set_source_group(TESTLEB "Source Files" dwarf_leb_test.c 
