diff --git a/libdwarf/CMakeLists.txt b/libdwarf/CMakeLists.txt
index 120f29d..6a9bd80 100644
--- a/libdwarf/CMakeLists.txt
+++ b/libdwarf/CMakeLists.txt
@@ -80,7 +80,7 @@ foreach(i RANGE ${targetCount})
             ${LIBELF_INCLUDE_DIRS})
 	msvc_posix(${target})
 
-	target_link_libraries(${target} PUBLIC ${LIBELF_LIBRARIES} z)
+	target_link_libraries(${target} PUBLIC ${CONAN_LIBS})
 	set_target_properties(${target} PROPERTIES OUTPUT_NAME dwarf)
 
 	install(TARGETS ${target}
